### PR TITLE
Improved Persistence

### PR DESCRIPTION
**Pull request by matthe815**
Added persistence to gil as well as TP. Begun adding support for elemental resistances based on deity.

### CODE REVIEW

File: src/world/Actor/Player.h [hunk 1]
```diff
@@ -736,6 +736,8 @@ namespace Sapphire::Entity
 
     void writeItem( ItemPtr pItem ) const;
 
+    void writeMoney(Common::CurrencyType type);
```
Comment (NotAdam) [inline src/world/Actor/Player.h:739]:
```suggestion
    void writeCurrencyItem( Common::CurrencyType type );
```

**Review by NotAdam — CHANGES_REQUESTED** [review]
can you fix the code style, it doesn't really match anything else (i.e. spaces)

File: src/world/Actor/PlayerInventory.cpp [hunk 2]
```diff
@@ -339,7 +339,7 @@ void Sapphire::Entity::Player::removeCurrency( Common::CurrencyType type, uint32
     currItem->setStackSize( 0 );
   else
     currItem->setStackSize( currentAmount - amount );
-  writeItem( currItem );
+  writeCurrencyItem(type);
```
Comment (NotAdam) [inline src/world/Actor/PlayerInventory.cpp:342]: spacing

File: src/world/Actor/PlayerSql.cpp [hunk 3]
```diff
@@ -696,6 +697,32 @@ bool Sapphire::Entity::Player::loadInventory()
...
+    "FROM charaitemcurrency " \
+    "WHERE CharacterId =  " + std::to_string(m_characterId) + " " \
+    "ORDER BY storageId ASC;");
+
+  while ( currencyRes->next() )
+  {
+    uint16_t storageId = currencyRes->getUInt16( 1 );
+    uint32_t money = currencyRes->getUInt64( 2 );
+
+    auto slot = static_cast< uint8_t >( static_cast<uint8_t>( CurrencyType::Gil ) - 1 );
```
Comment (NotAdam) [inline src/world/Actor/PlayerSql.cpp:713]:
```suggestion
    auto slot = static_cast< uint8_t >( static_cast< uint8_t >( CurrencyType::Gil ) - 1 );
```

File: src/world/Actor/PlayerSql.cpp [hunk 4]
```diff
@@ -696,6 +697,32 @@ bool Sapphire::Entity::Player::loadInventory()
     }
   }
 
+  auto currencyRes = db.query("SELECT storageId, "
```
Comment (NotAdam) [inline src/world/Actor/PlayerSql.cpp:700]: use fmt::format instead here actually

File: src/world/Actor/PlayerInventory.cpp [hunk 5]
```diff
@@ -557,6 +557,21 @@ void Sapphire::Entity::Player::writeItem( Sapphire::ItemPtr pItem ) const
...
+{
+  auto& db = Common::Service< Db::DbWorkerPool< Db::ZoneDbConnection > >::ref();
+
+  auto money = m_storageMap[ Currency ]->getItem( static_cast< uint16_t >( type ) - 1 )->getStackSize();
+
+  std::string query = "UPDATE charaitemcurrency SET ";
+
+  query += "container_" + std::to_string( static_cast< int16_t >( type ) - 1 ) + " = " + std::to_string( money );
+
+  query += " WHERE CharacterId = " + std::to_string( getCharacterId() ) + ";";
```
Comment (NotAdam) [inline src/world/Actor/PlayerInventory.cpp:570]: fmt::format again

**Review by NotAdam — CHANGES_REQUESTED** [review]

File: src/world/Actor/PlayerSql.cpp (hunk 4 above)
Comment (matthe815) [reply src/world/Actor/PlayerSql.cpp:700]: I was following the rest of the code in that regard, however I can rewrite most instances that I find if you'd like

Comment (NotAdam) [reply src/world/Actor/PlayerSql.cpp:700]: yeah a lot of that is pretty old - a lot of the db stuff is a bit antiquated and could be fixed but just fix these for now

I won't make you make fundamental changes outside of what bits you've touched, that'd be a dick move lol

Comment (matthe815) [reply src/world/Actor/PlayerSql.cpp:700]: Works for me. I'll have the fixes published in a few minutes.

**Review by NotAdam — APPROVED** [review]
cool - thanks for that, looks good.

btw you shouldn't need to edit CMakeLists for the script target, just rerun cmake and new scripts get added to the target

**Comment by matthe815** [comment]
I need to fix some build errors right quick, however the MakeList thing was done automatically by VS

**Comment by matthe815** [comment]
I intend to make a separate branch on my local repo and start adding all of the basic class quests.

**Pull request merged**